Source/JavaScriptCore/runtime/VMManager.cpp

```diff
@@ -43,6 +43,10 @@ VMManager& VMManager::singleton()
 VMThreadContext::VMThreadContext()
 {
     VM* vm = VM::fromThreadContext(this);
+    // Ensure that VM is not in-service yet. Since notifyVMConstruction has memory barrier (lock),
+    // if we are ensuring this condition here, concurrent threads will see this consistent state.
+    // Not that this bool field m_isInService is set before this VMThreadContext is initialized.
```

Comment:
I'm not quite sure what "Not that this bool field m_isInService is set" means (or is there a typo)?  Maybe you meant to say "Make sure m_isInService is initialized to false before VMThreadContext is initialized."  I think that would be clearer.

Reply:
Fixed.